packages/api/src/user.js

```diff
@@ -240,37 +240,62 @@ export async function userUploadsGet (request, env) {
     size = parsedSize
   }
 
-  let before = new Date()
+  let offset = 0
+  let page = 1
+  if (searchParams.has('page')) {
+    const parsedPage = parseInt(searchParams.get('page'))
+    if (isNaN(parsedPage) || parsedPage <= 0) {
+      throw Object.assign(new Error('invalid page number'), { status: 400 })
+    }
+    offset = (parsedPage - 1) * size
+    page = parsedPage
+  }
+
+  let before
   if (searchParams.has('before')) {
     const parsedBefore = new Date(searchParams.get('before'))
     if (isNaN(parsedBefore.getTime())) {
       throw Object.assign(new Error('invalid before date'), { status: 400 })
     }
-    before = parsedBefore
+    before = parsedBefore.toISOString()
+  }
+
+  let after
+  if (searchParams.has('after')) {
+    const parsedAfter = new Date(searchParams.get('after'))
+    if (isNaN(parsedAfter.getTime())) {
+      throw Object.assign(new Error('invalid after date'), { status: 400 })
+    }
+    after = parsedAfter.toISOString()
   }
 
   const sortBy = searchParams.get('sortBy') || 'Date'
```

Comment:
I'd be quite nice to validate theses, and return the right errors for unsupported values.
We use declaritive validators for the pin endpoint and we should probably port that to the uploads.
I think this is outside of the scope of the PR though, what do you think?

Reply:
Nice! Cool yeah agreed.
I added some validation below this however, now it'll validate on the API side if theres an unsupported `sortBy` param rather than borking the DB.